app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Toggle.kt

```diff
@@ -198,75 +214,113 @@ private fun DetailedToggle(
 
 @Composable
 private fun Toggle(enabled: Boolean, onClick: (Boolean) -> Unit, modifier: Modifier = Modifier) {
+  val density = LocalDensity.current
+  val positionalThreshold = { distance: Float -> distance * 0.3f }
+  val velocityThreshold = { with(density) { 100.dp.toPx() } }
+  val animationSpec = tween<Float>()
+  val decayAnimationSpec = rememberSplineBasedDecay<Float>()
+  val state = rememberSaveable(
```

Comment:
Great job with all this. I've never used this API myself, only seen it exist. What is tricky to get it all working? Are the docs good around it?

The fact that you need to kind duplicate passing everything to the saver and the state itself is a bit odd, perhaps we could even make our own `rememberAnchorDraggableState()` which takes in these four and calls the right rememberSaveable thing. For example like this https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/input/TextFieldState.kt;l=636-642?q=rememberTextFieldState which does the right remember with the right saver internally.